.github/workflows/ci.yml

```diff
@@ -0,0 +1,163 @@
+      - name: Cache Docker layers
+        uses: actions/cache@v2.1.5
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+      - name: Build test container
+        uses: docker/build-push-action@v2.4.0
+        with:
+          tags: cosi-driver-sample-test:ci
+          push: false
+          load: true
+          target: "test"
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache-new
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+      - name: Run tests
+        run: |
+          docker run \
+              -v $(pwd):/usr/src/cosi-driver-sample:ro \
+              cosi-driver-sample-test:ci
+      - name: Move cache
+        # Temporary fix
+        # See https://github.com/docker/build-push-action/blob/17822e4df435afa2aa88ec7d85b56ff55446ca36/docs/advanced/cache.md#github-cache
+        run: |
+          rm -rf /tmp/.buildx-cache
+          mv /tmp/.buildx-cache-new /tmp/.buildx-cache
+
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1.3.0
+      - name: Cache Docker layers
+        uses: actions/cache@v2.1.5
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+      - name: Extrace container image metadata
+        id: meta
+        uses: docker/metadata-action@v3.1.0
+        with:
+          images: cosi-driver-sample
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=sha
+            type=raw,value=ci
+      - name: Build
+        uses: docker/build-push-action@v2.4.0
+        with:
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          push: false
+          outputs: |
+            type=docker,dest=cosi-driver-sample.tar
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache-new
+      - name: Move cache
+        # Temporary fix
+        # See https://github.com/docker/build-push-action/blob/17822e4df435afa2aa88ec7d85b56ff55446ca36/docs/advanced/cache.md#github-cache
+        run: |
+          rm -rf /tmp/.buildx-cache
+          mv /tmp/.buildx-cache-new /tmp/.buildx-cache
+      - name: Archive artifacts
+        uses: actions/upload-artifact@v2.2.3
+        with:
+          name: container-images
+          path: cosi-driver-sample.tar
+
+  e2e:
+    name: E2E
+    needs:
+      - "build"
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
```

Comment:
I wanted to have *some* CI in the original repo. Since we don't use GitHub Actions in K8s, this will need to go elsewhere. All good. However, I'd like to keep this around until a replacement lands (at least) as an example of what such tests could (initially) look like.

Reply:
Removed.